src/Particle/Par_UpdateTracerParticle.cpp

```diff
@@ -193,24 +193,24 @@ void Par_UpdateTracerParticle( const int lv, const double TimeNew, const double
                   ParVel[d][ParID] = Vel_Temp    [d][p];
                }
 
-               ParTime[ParID] = TimeNew;
+               ParTime[ParID] = (real_par)TimeNew;
             }
 
 //          4.3 RK2 scheme (position only)
             else if ( amr->Par->IntegTracer == TRACER_INTEG_RK2 )
             {
 //             determine time-step
-               dt = (real)TimeNew - ParTime[ParID];
+               dt = (real)TimeNew - (real)ParTime[ParID];
 
 //             convert time-step for comoving
 #              ifdef COMOVING
-               if ( ParTime[ParID] == (real)TimeOld )    dt = dt_com;   // avoid redundant calculations
-               else                                      dt = Mis_dTime2dt( ParTime[ParID], dt );
+               if ( ParTime[ParID] == (real_par)TimeOld )    dt = dt_com;   // avoid redundant calculations
+               else                                          dt = Mis_dTime2dt( (const double)ParTime[ParID], (const double)dt );
 #              endif
 
                for (int d=0; d<3; d++)
                   InterpParPos[d][p] = ParPos[d][ParID] +
-                     (real)0.5*dt*( Vel_Temp[d][p] + ParVel[d][ParID] );
+                     (real_par)0.5*(real_par)dt*( Vel_Temp[d][p] + ParVel[d][ParID] );
```

Comment:
Remove `(real_par)` in front of `dt` if you agree upon declaring `dt` as `real_par`.